src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,507 @@
+        500: {"description": "Error encountered while submitting jobs."},
+        503: {"description": "Service unavailable."},
+    },
+    tags=["Ingestion"],
+    summary="submit jobs to the core nv ingestion service for processing with PDF splitting",
+    operation_id="submit_job_v2",
+)
+async def submit_job_v2(
+    request: Request, response: Response, job_spec: MessageWrapper, ingest_service: INGEST_SERVICE_T
+):
+    with tracer.start_as_current_span("http-submit-job-v2") as span:
+        try:
+            # Add custom attributes to the span
+            span.set_attribute("http.method", request.method)
+            span.set_attribute("http.url", str(request.url))
+            span.add_event("Submitting file for processing (V2)")
+
+            current_trace_id = span.get_span_context().trace_id
+            parent_job_id = trace_id_to_uuid(current_trace_id)
+
+            # Parse job spec
+            job_spec_dict = json.loads(job_spec.payload)
+
+            # Extract document type and payload from the proper structure
+            job_payload = job_spec_dict.get("job_payload", {})
+            document_types = job_payload.get("document_type", [])
+            payloads = job_payload.get("content", [])
+
+            # Check if this is a PDF that needs splitting
+            if document_types and payloads and document_types[0].lower() == "pdf":
+                # Decode the payload to check page count
+                pdf_content = base64.b64decode(payloads[0])
+                page_count = get_pdf_page_count(pdf_content)
+                pages_per_chunk = get_pdf_split_page_count()
+
+                # Split if the document has more pages than our chunk size
+                if page_count > pages_per_chunk:
+                    logger.info(
+                        "Splitting PDF with %s pages into chunks of %s",
+                        page_count,
+                        pages_per_chunk,
+                    )
+
+                    chunks = split_pdf_to_chunks(pdf_content, pages_per_chunk)
+
+                    subjob_ids: List[str] = []
+                    source_ids = job_payload.get("source_id", ["document.pdf"])
+                    source_names = job_payload.get("source_name", ["document.pdf"])
+                    original_source_id = source_ids[0] if source_ids else "document.pdf"
+                    original_source_name = source_names[0] if source_names else "document.pdf"
+
+                    try:
+                        parent_uuid = uuid.UUID(parent_job_id)
+                    except ValueError:
+                        logger.warning(
+                            "Parent job id %s is not a valid UUID; generating fallback namespace for subjobs",
+                            parent_job_id,
+                        )
+                        parent_uuid = uuid.uuid4()
+
+                    for chunk in chunks:
+                        chunk_number = chunk["chunk_index"] + 1
```

Comment:
Re: Juilo's monolithic function comment

This could be its own function to process each chunk, you could also get away with issuing all chunks asynchronously and then wait on all of them at the end of the loop.